WMF Framework/ReadingListsSyncOperation.swift

```diff
@@ -892,9 +893,12 @@ internal class ReadingListsSyncOperation: ReadingListsOperation {
         var entriesToDelete: [ReadingListEntry] = []
         for (readingListID, readingListEntriesByKey) in remoteReadingListEntriesByReadingListID {
             try autoreleasepool {
-                let localReadingListEntryFetch: NSFetchRequest<ReadingListEntry> = ReadingListEntry.fetchRequest()
-                localReadingListEntryFetch.predicate = NSPredicate(format: "list.readingListID == %@ && isDeletedLocally != YES", NSNumber(value: readingListID)) // this is != YES instead of == NO to match NULL values as well
-                let localReadingListEntries = try moc.fetch(localReadingListEntryFetch)
+
+                let localReadingListsFetch: NSFetchRequest<ReadingList> = ReadingList.fetchRequest()
+                localReadingListsFetch.predicate = NSPredicate(format: "readingListID == %@", NSNumber(value: readingListID))
+                let localReadingLists = try moc.fetch(localReadingListsFetch)
+                let localReadingListEntries = localReadingLists.first?.entries?.filter { !$0.isDeletedLocally } ?? []
```

Comment:
@joewalsh You were right, the problem was much more subtle. 
It was so subtle that I still have no idea why it occured.

Apparently fetching `ReadingListEntry` by `list` relationship:

`localReadingListEntryFetch.predicate = NSPredicate(format: "list.readingListID == %@ && isDeletedLocally != YES", NSNumber(value: readingListID))`

Doesn't bring expected results. Specifically: the first fetch brings `0` entries while there definitely are entries and:

`localReadingListsFetch.predicate = NSPredicate(format: "readingListID == %@", NSNumber(value: readingListID))`

Does indeed work and fetches the expected entries. It's weird as these queries should have the same result. I'll try to debug some more to find out what's the reason behind it.

Reply:
This is very puzzling. I've spent another hour debugging this and the result is the same: `list.readingListID == %@` sometimes returns `0` results even though clearly there should be results. It doesn't seem to do with faulting too.

The new query fixes the issue.

Reply:
If we don't include `&& isDeletedLocally != YES` in the predicate, lists previously deleted locally will show up. You can test it by logging in, adding a couple of articles to list Foo, logging out (while keeping articles on your device), creating another list Bar (can be empty) && deleting it, logging in and enabling sync. We'll see both Foo and Bar, and we only care about Foo since Bar was deleted.